modules/apps/frontend-js/frontend-js-dependencies-web/src/main/java/com/liferay/frontend/js/dependencies/web/internal/servlet/taglib/JSDependenciesInitTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.dependencies.web.internal.servlet.taglib;
+
+import com.liferay.frontend.js.loader.modules.extender.esm.ESImportUtil;
+import com.liferay.portal.kernel.servlet.taglib.BaseDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.aui.JSFragment;
+import com.liferay.portal.kernel.servlet.taglib.aui.ScriptData;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilder;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilderFactory;
+
+import java.io.IOException;
+
+import java.util.Arrays;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(service = DynamicInclude.class)
+public class JSDependenciesInitTopHeadDynamicInclude
```

Comment:
I know we use this pattern of injecting JS files from a dynamic include in more places, but it goes against performance because it doesn't allow pruning unused stuff. 

Meaning this will be executed in public sites that don't use date-fns at all lowering the performance for no benefit.

Reply:
Any ideas on a way to dynamically add this only if date-fns is used? And it would need to be before date-fns is used.

This was the only way I could really think of setting the default locale without having to manually change it in every instance of date-fns